specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/cloudhsm.json

```diff
@@ -0,0 +1,1470 @@
+          },
+          {
+            "$ref": "#/parameters/cloudHsmClusterName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "restoreRequestProperties",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RestoreRequestProperties"
+            },
+            "description": "Restore Operation Required properties"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. Cloud HSM Cluster Restore operation API finished synchronously with Success",
+            "schema": {
+              "$ref": "#/definitions/RestoreResult"
+            },
+            "headers": {
+              "x-ms-request-id": {
+                "description": "A unique ID for the current operation, service generated. All the resource providers must return this value in the response headers to facilitate debugging.",
+                "type": "string"
+              }
+            }
+          },
+          "202": {
+            "description": "Cloud HSM Cluster Restore operation started",
+            "headers": {
+              "Retry-After": {
+                "description": "The recommended number of seconds to wait before calling the URI specified in Azure-AsyncOperation.",
+                "type": "integer",
+                "format": "int32"
+              },
+              "Azure-AsyncOperation": {
+                "description": "The URI to poll for completion status.",
+                "type": "string"
+              },
+              "Location": {
+                "description": "The URI to poll for completion status",
+                "type": "string"
+              },
+              "x-ms-request-id": {
+                "description": "A unique ID for the current operation, service generated. All the resource providers must return this value in the response headers to facilitate debugging.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "CloudHsmCluster_Restore_MaximumSet_Gen": {
+            "$ref": "./examples/CloudHsmCluster_RequestOrValidate_Restore_MaximumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HardwareSecurityModules/cloudHsmClusters/{cloudHsmClusterName}/restore/{jobId}/pending": {
+      "get": {
+        "tags": [
+          "Returns Restore Operation Status"
+        ],
+        "operationId": "CloudHsmClustersRestoreStatus_Get",
```

Comment:
>CloudHsmClustersRestoreStatus_Get [](http://example.com/codeflow?start=24&length=33)

I think this URI shape would be interpreted by ARM as a 'list' operation to 'list pendings by restore job'

Reply:
If its supposed to be a point get, would it make sense about if instead it were 

GET "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HardwareSecurityModules/cloudHsmClusters/{cloudHsmClusterName}/restoreJobResults/{jobId}"

?

Reply:
Updated the API to this:     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HardwareSecurityModules/cloudHsmClusters/{cloudHsmClusterName}/restoreOperationStatus/{jobId}": {
Please let me know if this violates the linter rules, thanks !

Reply:
updated the API and linter validations went away. thanks !